packages/flutter/lib/src/painting/rounded_rectangle_border.dart

```diff
@@ -0,0 +1,298 @@
+      return new _RoundedRectangleToCircleBorder(
+        side: BorderSide.lerp(side, b.side, t),
+        borderRadius: BorderRadius.lerp(borderRadius, b.borderRadius, t),
+        circleness: circleness * (1.0 - t),
+      );
+    }
+    if (b is CircleBorder) {
+      return new _RoundedRectangleToCircleBorder(
+        side: BorderSide.lerp(side, b.side, t),
+        borderRadius: borderRadius,
+        circleness: circleness + (1.0 - circleness) * t,
+      );
+    }
+    if (b is _RoundedRectangleToCircleBorder) {
+      return new _RoundedRectangleToCircleBorder(
+        side: BorderSide.lerp(side, b.side, t),
+        borderRadius: BorderRadius.lerp(borderRadius, b.borderRadius, t),
+        circleness: ui.lerpDouble(circleness, b.circleness, t),
+      );
+    }
+    return super.lerpTo(b, t);
+  }
+
+  Rect _adjustRect(Rect rect) {
+    if (circleness == 0.0 || rect.width == rect.height)
+      return rect;
+    if (rect.width < rect.height) {
+      final double delta = circleness * (rect.height - rect.width) / 2.0;
+      return new Rect.fromLTRB(
+        rect.left,
+        rect.top + delta,
+        rect.right,
+        rect.bottom - delta,
+      );
+    } else {
+      final double delta = circleness * (rect.width - rect.height) / 2.0;
+      return new Rect.fromLTRB(
+        rect.left + delta,
+        rect.top,
+        rect.right - delta,
+        rect.bottom,
+      );
+    }
+  }
+
+  BorderRadius _adjustBorderRadius(Rect rect) {
+    if (circleness == 0.0)
+      return borderRadius;
+    return BorderRadius.lerp(borderRadius, new BorderRadius.circular(rect.shortestSide / 2.0), circleness);
+  }
+
+  @override
+  Path getInnerPath(Rect rect, { TextDirection textDirection }) {
+    return new Path()
+      ..addRRect(_adjustBorderRadius(rect).toRRect(_adjustRect(rect)).deflate(side.width));
+  }
+
+  @override
+  Path getOuterPath(Rect rect, { TextDirection textDirection }) {
+    return new Path()
+      ..addRRect(_adjustBorderRadius(rect).toRRect(_adjustRect(rect)));
+  }
+
+  @override
+  void paint(Canvas canvas, Rect rect, { TextDirection textDirection }) {
+    switch (side.style) {
+      case BorderStyle.none:
+        break;
+      case BorderStyle.solid:
+        final double width = side.width;
+        if (width == 0.0) {
+          canvas.drawRRect(_adjustBorderRadius(rect).toRRect(_adjustRect(rect)), side.toPaint());
+        } else {
+          final RRect outer = _adjustBorderRadius(rect).toRRect(_adjustRect(rect));
+          final RRect inner = outer.deflate(width);
+          final Paint paint = new Paint()
+            ..color = side.color;
+          canvas.drawDRRect(outer, inner, paint);
+        }
+    }
+  }
+
+  @override
+  bool operator ==(dynamic other) {
+    if (runtimeType != other.runtimeType)
+      return false;
+    final _RoundedRectangleToCircleBorder typedOther = other;
+    return side == typedOther.side
+        && borderRadius == typedOther.borderRadius
```

Comment:
Are there two extra spaces of indent here?

Reply:
this is abarth style to vertically align the subexpressions